### PR TITLE
Added `+` and `-` as postfix operators, for use with Plato

### PR DESCRIPTION
**Pull request by jrbeaumont**

### CODE REVIEW

**Comment by jrbeaumont** [comment]
@snowleopard I have made the changes to `boolean`, so when the PR for process-mining is complete, we can update this too, then I will need to update the commit in Plato, but everything will then work together nicely :)

File: src/Tuura/Boolean/Parser.hs [hunk 1]
```diff
@@ -33,10 +33,12 @@ instance Monad Expr where
...
 parseExpr = parse expr ""
   where expr = buildExpressionParser operators term <?> "compound expression"
-        term = parens expr <|> variable <?> "full expr ession"
+        term = parens expr <|> variable <?> "full expression"
         operators = [ [Prefix (string "NOT" >> spaces >> return Not),
                        Prefix (string "!" >> return Not),
-                       Postfix (string "'" >> return Not)]
+                       Postfix (string "'" >> spaces >> return Not),
+                       Postfix (string "-" >> spaces >> return Not),
+                       Postfix (string "+" >> spaces >> return (Not . Not))]
```
Comment (snowleopard) [inline src/Tuura/Boolean/Parser.hs:41]: `Not . Not` looks mysterious. Did you mean `id`? :-)

**Review by snowleopard — COMMENTED** [review]
Looks fine! Just simplify the `Not . Not` bit and I'll merge.

**Comment by ARandomOWL** [comment]
I don't know what the use-case is for these operators, but I'm not sure it's a good decision in the long-term to bundle them into the main parser like this. I can see confusing conflicts occurring when parsing human input. Perhaps some conversion of format can occur between the tools instead?

**Comment by snowleopard** [comment]
@ARandomOWL Yes, I agree that this is not the best solution. 

@jrbeaumont Do you think we could introduce an intermediate conversion that simply erases all `+` and replaces `-` with `'`?

**Comment by jrbeaumont** [comment]
It is not the best, a conversion is likely possible on the PGminer end, where as it's generating the Boolean function, it finds any variables ending in `+` or `-` and deals with them accordingly. Let me see what I can do, then we can close this PR.

**Comment by jrbeaumont** [comment]
The `+` and `-` is now removed or replaced by PGminer. This PR can be closed.

**Pull request closed without merging**